microsoft-edge/webview2/gettingstarted/winui.md

```diff
@@ -0,0 +1,291 @@
+
+In `MainWindow.xaml.cs`, modify the constructor as shown below and add the `EnsureHttps` function.  
+
+```csharp
+public MainWindow()
+{
+    InitializeComponent();
+    MyWebView.NavigationStarting += EnsureHttps;
+}
+
+private void EnsureHttps(WebView2 sender, WebView2NavigationStartingEventArgs args)
+{
+    String uri = args.Uri;
+    if (!uri.StartsWith("https://"))
+    {
+        args.Cancel = true;
+    }
+    else
+    {
+        addressBar.Text = uri;
+    }
+}
+```  
+
+In the constructor, EnsureHttps is registered as the event handler on the `NavigationStarting` event on the WebView2 control.  
+
+Press `F5` to build and run your project.  Confirm that when navigating to an HTTP site, the WebView **remains unchanged**.  However, the WebView navigates to HTTPS sites.  
+
+## Step 5 - Scripting  
+
+You may use host applications to inject JavaScript code into WebView2 controls at runtime.  The injected JavaScript applies to all new top level documents and any child frames until the JavaScript is removed.  The injected JavaScript is run after creation of the global object, and before any other script included in the HTML document is run.  
+
+You can use scripting to alert the user when navigating to a non-HTTPS site.  Modify the `EnsureHttps` function so that it injects script into the web content using the [ExecuteScriptAsync](../reference/wpf/0-9-515/microsoft-web-webview2-wpf-webview2.md#executescriptasync) method.  
+
+```csharp
+private void EnsureHttps(WebView2 sender, WebView2NavigationStartingEventArgs args)
+{
+    String uri = args.Uri;
+    if (!uri.StartsWith("https://"))
+    {
+        MyWebView.ExecuteScriptAsync($"alert('{uri} is not safe, try an https link')");
+        args.Cancel = true;
+    }
+    else
+    {
+        addressBar.Text = uri;
+    }
+}
+```  
+
+Press `F5` to build and run your project.  Confirm that the application displays an alert when you navigate to a site that does not use HTTPS.  
+
+:::image type="complex" source="./media/winui-gettingstarted-script.png" alt-text="HTTPS":::
+   HTTPS
+:::image-end:::  
+
+
+
+
+**Congratulations, you built your first WebView2 app!**
+  
+## Future Plans  
+
+Our team is currently working on building out our APIs to provide full WebView2 functionality! For more information of the current state of our API click [here](https://github.com/microsoft/microsoft-ui-xaml-specs/blob/master/active/WebView2/WebView2_spec.md). 
+
+> [!NOTE]
+> This API may ship before the WinRT CoreWebView2 object, meaning functionality of the Webview2 would be limited to the APIs included in the above document for some interim period of time. 
+
+
+*   For a comprehensive example of WebView2 capabilities, see [WebView2Samples repo](https://github.com/MicrosoftEdge/WebView2Samples) on GitHub.  
+*   For more information about  WebView2, see [WebView2 Resources](../index.md#next-steps).  
+
```

Comment:
Add a "Next Steps" section here to keep users engaged with content:
Point to WinUI Docs, our Sample Apps, etc. whatever you think is best!

Reply:
I renamed the Future Plans to Next Steps - they are essentially doing the same thing!